### PR TITLE
Milestone 2 Initilize Redux

### PR DESCRIPTION
**Pull request by FrancoRosa**
Hello TSE
In this milestone I manage to:

- Prepare the directory structure
- Create the book reducer
- Implement the App component
- Implement the BooksList component
- Implement the Book component
- Implement the BooksForm component
- Define the Redux actions and reducers
- Initialize the Redux store

Please give me some feedback.

### CODE REVIEW

File: src/components/BooksForm.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
```
Comment (bellom) [inline src/components/BooksForm.js]: Let's do some rearrangement (create `container` folder and put `BookForm.js` in it)
React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory (presentational components) you have components that are not connected to the Redux store, and on your containers directory, you have components that are connected to the Redux store. https://redux.js.org/basics/usage-with-react#presentational-and-container-components 

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

File: src/components/BooksList.js [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
```
Comment (bellom) [inline src/components/BooksList.js]: Let's do some rearrangement (create `container` folder and put `BookList.js` in it)
React bindings for Redux separate presentational components from container components. This approach can make your app easier to understand and allow you to more easily reuse components. In your components directory (presentational components) you have components that are not connected to the Redux store, and on your containers directory, you have components that are connected to the Redux store. https://redux.js.org/basics/usage-with-react#presentational-and-container-components 

image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

File: src/components/BooksList.js [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        {books.map(book => (
+          <Book key={book.id} book={book} />
+        ))}
+      </tbody>
+    </table>
+  </div>
+);
+
+BooksList.propTypes = {
+  books: PropTypes.arrayOf(Book.propTypes).isRequired,
```
Comment (bellom) [inline src/components/BooksList.js]: Using [PropTypes.shape](https://dev.to/cesareferrari/how-to-specify-the-shape-of-an-object-with-proptypes-3c56) here will allow you to set a type check that is more accurate.

File: src/reducers/index.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    id: Math.random(),
+    title: 'Some guy bio',
+    category: 'Biography',
+  },
+  {
+    id: Math.random(),
+    title: 'Watch out, the return',
+    category: 'Horror',
+  },
+];
```
Comment (bellom) [inline src/reducers/index.js]: `initialState` shoud be inside `src/index.js`
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

File: src/reducers/books.js [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    id: Math.random(),
+    title: 'Some guy bio',
+    category: 'Biography',
+  },
+  {
+    id: Math.random(),
+    title: 'Watch out, the return',
+    category: 'Horror',
+  },
+];
```
Comment (bellom) [inline src/reducers/books.js]: `initialState` shoud be inside `src/index.js`
image
https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/

**Review by bellom — CHANGES_REQUESTED** [review]
## Hi @FrancoRosa  :clap: 

### Awesome job so far!! :+1: 

#### You have done wonderfully well, you passed `almost` all the checklist for this project :+1: just a few issues left to work on............ :slightly_smiling_face: 

# Status: Changes Required! ♻️ 

**Kindly check out my comments below;**

### ♻️ Kindly re-submit for review after making all those changes :slightly_smiling_face: Thumps -up await you.....

**Best,**

**Oluwaseun Bello**
**_Slack Handle - @sean bellom_**

**Comment by FrancoRosa** [comment]
Thank you @bellom for your kind review and suggestions
I did the required changes

**Review by wathigo — APPROVED** [review]
### Approved :heavy_check_mark: 
Great job implementing the suggested changes.

**You may proceed to the next milestone.**

**Happy Hacking :man_technologist: 
Simon Wathigo.**

**Pull request merged**